frontend/packages/kubevirt-plugin/src/components/vm-templates/vm-template-create-yaml.tsx

```diff
@@ -22,13 +23,38 @@ const CreateVMTemplateYAMLConnected = connectToPlural(
       }
       return <ErrorPage404 />;
     }
+    const [selectedTemplate, setTemplate] = React.useState(null);
+    const [templateLabel, setTemplateLabel] = React.useState(null);
 
     const template = VMTemplateYAMLTemplates.getIn(['vm-template']);
     const obj = safeLoad(template);
     obj.kind = kindObj.kind;
     obj.metadata = obj.metadata || {};
     obj.metadata.namespace = match.params.ns || 'default';
 
+    if (templateLabel) {
+      obj.objects[0].metadata.labels['vm.kubevirt.io/template'] = templateLabel;
+    }
+
+    if (!selectedTemplate) {
+      k8sGet(TemplateModel, '', 'openshift')
+        .then((templates) => {
+          setTemplate(
+            templates.items.filter(
+              (item) =>
```

Comment:
can we make a request which matches the labels we need? Similar to what we do in Firehose with `matchLabels`. 

n we check the annotations for the fedora first and/or fallback to the name if nothing is found?